[2025-04-22T12:45:10.936+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-04-22T12:45:10.991+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: final_project_BBDL.LandingToBronze manual__2025-04-22T12:45:09.519540+00:00 [queued]>
[2025-04-22T12:45:11.000+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: final_project_BBDL.LandingToBronze manual__2025-04-22T12:45:09.519540+00:00 [queued]>
[2025-04-22T12:45:11.001+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-04-22T12:45:11.063+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): LandingToBronze> on 2025-04-22 12:45:09.519540+00:00
[2025-04-22T12:45:11.069+0000] {logging_mixin.py:188} WARNING - /home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=174) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-04-22T12:45:11.070+0000] {standard_task_runner.py:63} INFO - Started process 175 to run task
[2025-04-22T12:45:11.069+0000] {standard_task_runner.py:90} INFO - Running: ['airflow', 'tasks', 'run', 'final_project_BBDL', 'LandingToBronze', 'manual__2025-04-22T12:45:09.519540+00:00', '--job-id', '2', '--raw', '--subdir', 'DAGS_FOLDER/project_solution.py', '--cfg-path', '/tmp/tmpnhbiiboi']
[2025-04-22T12:45:11.071+0000] {standard_task_runner.py:91} INFO - Job 2: Subtask LandingToBronze
[2025-04-22T12:45:11.215+0000] {logging_mixin.py:188} WARNING - /home/airflow/.local/lib/python3.12/site-packages/airflow/settings.py:195 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2025-04-22T12:45:11.301+0000] {task_command.py:426} INFO - Running <TaskInstance: final_project_BBDL.LandingToBronze manual__2025-04-22T12:45:09.519540+00:00 [running]> on host 88554498a81a
[2025-04-22T12:45:11.517+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='final_project_BBDL' AIRFLOW_CTX_TASK_ID='LandingToBronze' AIRFLOW_CTX_EXECUTION_DATE='2025-04-22T12:45:09.519540+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-04-22T12:45:09.519540+00:00'
[2025-04-22T12:45:11.518+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-04-22T12:45:11.663+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-04-22T12:45:11.664+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/landing_to_bronze.py", line 23, in main
    .getOrCreate()
     ^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/sql/session.py", line 497, in getOrCreate
    sc = SparkContext.getOrCreate(sparkConf)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/context.py", line 515, in getOrCreate
    SparkContext(conf=conf or SparkConf())
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/context.py", line 201, in __init__
    SparkContext._ensure_initialized(self, gateway=gateway, conf=conf)
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/context.py", line 436, in _ensure_initialized
    SparkContext._gateway = gateway or launch_gateway(conf)
                                       ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/java_gateway.py", line 97, in launch_gateway
    proc = Popen(command, **popen_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/local/lib/python3.12/subprocess.py", line 1955, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/home/olena/spark/./bin/spark-submit'
[2025-04-22T12:45:11.702+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=final_project_BBDL, task_id=LandingToBronze, run_id=manual__2025-04-22T12:45:09.519540+00:00, execution_date=20250422T124509, start_date=20250422T124510, end_date=20250422T124511
[2025-04-22T12:45:11.728+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 2 for task LandingToBronze ([Errno 2] No such file or directory: '/home/olena/spark/./bin/spark-submit'; 175)
[2025-04-22T12:45:11.767+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-04-22T12:45:11.811+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-04-22T12:45:11.827+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
